source/common/network/common_connection_filter_states.h

```diff
@@ -0,0 +1,38 @@
+#pragma once
+
+#include "envoy/common/execution_context.h"
+#include "envoy/network/connection.h"
+#include "envoy/stream_info/filter_state.h"
+
+namespace Envoy {
+namespace Network {
+
+static constexpr absl::string_view kConnectionExecutionContextFilterStateName =
+    "envoy.network.connection_execution_context";
+
+// ConnectionExecutionContextFilterState is an optional connection-level filter state that goes by
+// the name kConnectionExecutionContextFilterStateName. It owns a ExecutionContext, whose
+// activate/deactivate methods will be called when a thread starts/finishes running code on behalf
+// of the corresponding connection.
+class ConnectionExecutionContextFilterState : public Envoy::StreamInfo::FilterState::Object {
+public:
+  // It is safe, although useless, to set execution_context to nullptr.
+  explicit ConnectionExecutionContextFilterState(
+      std::unique_ptr<ExecutionContext> execution_context)
+      : execution_context_(std::move(execution_context)) {}
+
+  const ExecutionContext* executionContext() const { return execution_context_.get(); }
```

Comment:
I think you can return `ExecutionContext*` directly here, without the `const` qualifier. The final function would be:

```
ExecutionContext* executionContext() const { return execution_context_.get(); }
```

Then, you only need one getConnectionExecutionContext method and needn't using the `const_cast`.

Reply:
Oh, for some reason I thought `const_cast` is unavoidable, but clearly I was wrong.  Changed to one getConnectionExecutionContext as you suggested.

Thank you for the great suggestion!